src/platforms/hosted/jlink.c

```diff
@@ -151,91 +175,24 @@ bool jlink_transfer_swd(
 	/* NOLINTNEXTLINE(readability-suspicious-call-argument) */
 	return jlink_transfer(clock_cycles, dir, data_in, data_out);
 }
 
-static bool jlink_print_version(void)
-{
-	uint8_t len_str[2];
-	if (jlink_simple_query(JLINK_CMD_GET_VERSION, len_str, sizeof(len_str)) < 0)
-		return false;
-	uint8_t version[0x70];
-	bmda_usb_transfer(info.usb_link, NULL, 0, version, sizeof(version));
-	version[0x6f] = '\0';
-	DEBUG_INFO("%s\n", version);
-	return true;
-}
-
-static bool jlink_query_caps(void)
-{
-	uint8_t caps[4];
-	if (jlink_simple_query(JLINK_CMD_GET_CAPABILITIES, caps, sizeof(caps)) < 0)
-		return false;
-	jlink_caps = read_le4(caps, 0);
-	DEBUG_INFO("Caps %" PRIx32 "\n", jlink_caps);
-
-	if (jlink_caps & JLINK_CAP_GET_HW_VERSION) {
-		uint8_t version[4];
-		if (jlink_simple_query(JLINK_CMD_GET_ADAPTOR_VERSION, version, sizeof(version)) < 0)
-			return false;
-		DEBUG_INFO("HW: Type %u, Major %u, Minor %u, Rev %u\n", version[3], version[2], version[1], version[0]);
-	}
-	return true;
-}
-
-static bool jlink_query_speed(void)
-{
-	uint8_t data[6];
-	if (jlink_simple_query(JLINK_CMD_GET_ADAPTOR_FREQS, data, sizeof(data)) < 0)
-		return false;
-	jlink_freq_khz = read_le4(data, 0) / 1000U;
-	jlink_min_divisor = read_le2(data, 4);
-	DEBUG_INFO("Emulator speed %ukHz, minimum divisor %u%s\n", jlink_freq_khz, jlink_min_divisor,
-		(jlink_caps & JLINK_CAP_GET_SPEEDS) ? "" : ", fixed");
-	return true;
-}
-
-static bool jlink_print_interfaces(void)
-{
-	uint8_t active_if[4];
-	uint8_t available_ifs[4];
-
-	if (jlink_simple_request(JLINK_CMD_TARGET_IF, JLINK_IF_GET_ACTIVE, active_if, sizeof(active_if)) < 0 ||
-		jlink_simple_request(JLINK_CMD_TARGET_IF, JLINK_IF_GET_AVAILABLE, available_ifs, sizeof(available_ifs)) < 0)
-		return false;
-	++active_if[0];
-	jlink_interfaces = available_ifs[0];
-
-	if (active_if[0] == JLINK_IF_SWD)
-		DEBUG_INFO("SWD active");
-	else if (active_if[0] == JLINK_IF_JTAG)
-		DEBUG_INFO("JTAG active");
-	else
-		DEBUG_INFO("No interfaces active");
-
-	const uint8_t other_interface = available_ifs[0] - active_if[0];
-	if (other_interface)
-		DEBUG_INFO(", %s available\n", other_interface == JLINK_IF_SWD ? "SWD" : "JTAG");
-	else
-		DEBUG_INFO(", %s not available\n", active_if[0] + 1U == JLINK_IF_SWD ? "JTAG" : "SWD");
-	return true;
-}
-
-static bool jlink_info(void)
-{
-	return jlink_print_version() && jlink_query_caps() && jlink_query_speed() && jlink_print_interfaces();
-}
-
 /*
  * Try to claim the debugging interface of a J-Link adaptor.
  * On success this copies the endpoint addresses identified into the
  * usb_link_s sub-structure of bmp_info_s (info.usb_link) for later use.
  * Returns true for success, false for failure.
+ *
+ * Note: Newer J-Links use 2 bulk endpoints, one for "IN" (EP1) and one
+ * for "OUT" (EP2) communication whereas old J-Links (V3, V4) only use
+ * one endpoint (EP1) for "IN" and "OUT" communication.
+ * Presently we only support the newer J-Links with 2 bulk endpoints.
```

Comment:
How difficult would it be to detect and configure for the older style? If that might even be vaguely useful..

Reply:
Probably fairly straight forward, at least the raw protocol, but I believe it will be extremely unlikely that one is found in the wild, I also don't have one to test, and to pile on, I think older J-Links might not have some of the newer CMD_IO commands we are using

Perhaps we could add a warning when a single endpoint probe is detected, but I'm starting to burnout on J-Link code so I won't

Reply:
No worries, this can be a later enhancement and a low priority one at that then.